File: bench/04-primes.py
def is_prime(n: int) -> bool:
    if n <= 1:
        return False
    return all(n % i != 0 for i in range(2, n))

def primes_up_to(n: int) -> list[int]:
    return [x for x in range(2, n + 1) if is_prime(x)]

def nth_prime(n: int) -> int:
    return primes_up_to(n * 20)[0]

File: bench/04-primes.java
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.LongStream;

final class Primes {
  public static boolean isPrime(long n) {
    return n > 1 && LongStream.range(2, n).allMatch(i -> n % i != 0);
  }

  public static List<Long> primesUpTo(long n) {
    return LongStream.range(2, n + 1).filter(Primes::isPrime).boxed().collect(Collectors.toList());
  }

  public static long nthPrime(long n) {
    return primesUpTo(n * 20).get(0);
  }
}
